pmd-test/src/main/java/net/sourceforge/pmd/ant/AbstractAntTestHelper.java

```diff
@@ -46,6 +50,11 @@ public void setUp() {
         // initialize Ant
         buildRule.configureProject(pathToTestScript + separator + antTestScriptFilename);
 
+        // Each test case gets one temp file name, accessible with ${tmpfile}
+        // The file doesn't exist before the test is executed
+        Path tmpFilePath = FileUtils.getTempDirectory().toPath().resolve(RandomStringUtils.randomAlphabetic(30)).toAbsolutePath();
+        buildRule.getProject().setProperty("tmpfile", tmpFilePath.toString());
```

Comment:
you can use `TemporaryFolder` (junit rule) to have a unique folder created per test, and have it automatically cleaned up afterwards.